wordpress-mobile/WordPress-iOS
# [Site Creation] Segments Screen

Early pull request, taking the implementation of  #10322 as an excuse to add most, if not all, of the basic scaffolding necessary to build the UI of the new Site Creation flow.

In this PR:
- First implementation of a `WizardStep`. 
- First implementation of the View Controller that would render all the steps headers
- First implementation of the View Controller that renders the Segments Screen, plus cell and datasource
- A few model objects supporting the implementation.
- A builder object, that we'd use to propagate the state of the process (track the data collected from user input/selection)
- The container view controller that we'd use to render the wizard
- A class that puts everything together (create steps, container, set them up for collaboration) and launches the wizard

To have in mind:
- The actual UI of the Segments Screen is (obviously) a draft
- I can not make the `SiteSegmentsDataSource` to run. I assume it will have to be because of the way I am bootstrapping the UI, but I am at that moment when the trees do not let me see the forest.
- I didn't add tests yet. I wanted to get this out as soon as possible. 

To test:
- Make sure the feature flag is active.
- Navigate to the My Sites tab, and tap the plus button in the navigation bar.
- Tap "Enhanced site creation" 
![simulator screen shot - iphone xr - 2018-10-31 at 23 03 09](https://user-images.githubusercontent.com/2722505/47835460-442a7700-dd61-11e8-8245-817c2a96e6f7.png)
